src/rating/rating.js

```diff
@@ -12,6 +12,13 @@ angular.module('ui.bootstrap.rating', [])
   this.init = function(ngModelCtrl_) {
     ngModelCtrl = ngModelCtrl_;
     ngModelCtrl.$render = this.render;
+    
+    ngModelCtrl.$formatters.push(function(value) {
+      if (typeof value !== 'undefined' && value % 1 !== 0) {
```

Comment:
Is there any reason why we can't just do
<pre>
if (angular.isNumber(value) && value === value) {
</pre>
here?  It seems you are trying to test for whether it is a number, this would be a better check.